medicines/doc-index-updater/src/service_bus_client/mod.rs

```diff
@@ -63,15 +64,33 @@ impl Error for RetrieveFromQueueError {
     }
 }
 
+pub struct RetrievedMessage<T> {
+    pub message: T,
+    peek_lock: PeekLockResponse,
+}
+
+impl<T> RetrievedMessage<T> {
+    pub async fn remove(&self) -> Result<String, anyhow::Error> {
+        let queue_removal_result = self.peek_lock.delete_message().await.map_err(|e| {
+            tracing::error!("{:?}", e);
+            anyhow!("Queue Removal Error")
+        });
```

Comment:
Does this need a `?` so that we return on error?

Reply:
two lines later we return the whole result - we could `?` here instead and then in the tracing, we'd only see the success. That makes a bit more sense to me. I'll raise the PR for that